### ISSUE TITLE
feat: Add CITL export for OIP integration

### ISSUE THREAD

**Issue by noahgift** [paiml/ruchy#165]
## Summary

Add Compiler-In-The-Loop (CITL) export support to enable integration with organizational-intelligence-plugin (OIP) for defect pattern analysis and training data generation.

## Background

OIP uses compiler diagnostics as ground-truth labels for ML-based defect classification. Depyler already implements CITL export. This ticket adds the same capability to ruchy (Ruby→Rust transpiler).

### CLI Flag
```bash
ruchy transpile input.rb --citl-export diagnostics.json
```

### Export Format
```rust
#[derive(Serialize)]
pub struct CitlExport {
    pub source_file: String,
    pub error_code: Option<String>,      // e.g., "E0308"
    pub clippy_lint: Option<String>,     // e.g., "clippy::unwrap_used"
    pub level: String,                   // "error" | "warning"
    pub message: String,
    pub oip_category: Option<String>,    // Pre-classified if known
    pub confidence: f32,
    pub span: Option<SpanInfo>,
    pub suggestion: Option<SuggestionInfo>,
    pub timestamp: i64,
}
```

### Ruby-Specific Considerations
- Map Ruby type errors to Rust equivalents
- Capture method resolution failures
- Track duck-typing → trait conversion issues

## Effort
~2-4 hours

## Related
- OIP CITL spec: `organizational-intelligence-plugin/src/citl.rs`
- Depyler implementation: `depyler/crates/depyler-core/src/citl.rs`
